### PR TITLE
added dataset methods, removed Pre-API

### PR DESCRIPTION
**Pull request by JasonD94**
Adds a method to return a vector of data off of iSENSE given a dataset name and a field name. Basically gets one "row of data" off of iSENSE.
I also removed the Pre-API folder since that is mostly old starter code I used for some of the API methods (early GET / POSTs using curl).
Also not sure if I should leave the examples in here or remove them since they are in the Teaching repo as well.

- Ignore picojson.h, that got added by mistake.

@bdonald25 @stowellm

### CODE REVIEW

**Comment by stowellm** [comment]
Looks good.  Just a question about the examples though: are there any other API usage examples in that /Examples subdirectory?  Will there be?  Duplicate code is bad, so I would assume it should only live in one repository, and my assumption is that that would be the Teaching repository.

**Comment by JasonD94** [comment]
I'll remove it from here then, and just keep the examples / Apps in the Teaching Repo.
Then in the Teaching Repo I'll remove the API files since they can live here.
In the future I'll write a bash script that people can use to just git clone the two repos + pull the picojson header file that the C++ code uses.

**Comment by stowellm** [comment]
Awesome - good to pull.

**Comment by JasonD94** [comment]
Made the suggestions from the Teaching repo - added constants for the format_upload_string function. Also made "int key" -> "int post_type".

File: C++/API/API.cpp [hunk 1]
```diff
@@ -1158,30 +1324,30 @@ bool iSENSE::append_email_byName(string dataset_name)
...
   // Add the title + the correct formatting
   upload_data["title"] = value(title);
 
-  // This is now a switch. Any future API methods can be added here.
-  switch(key)
+  // This is now a switch. Future API methods can be added here.
+  switch(post_type)
   {
-    case 1:
+    case Post_Key:
```
Comment (stowellm) [inline C++/API/API.cpp]: Constants are usually defined as all uppercase, i.e. `POST_KEY`.  You already did this for `CURL_ERROR`, i.e.

**Comment by stowellm** [comment]
One comment then looks good.  It's hard to judge from just looking at this PR how the repository will be structured, so I'm tempted to say once the above comment is addressed, let's pull it and then I'll glance over the repository structure to make sure I don't find any duplicates and that everything is in place.

**Comment by JasonD94** [comment]
Fixed that one issue with the constants.

**Pull request merged**